### PR TITLE
CQ-4348824 Support of Repeatable Panel inside Repeatable Panel.

### PR DESCRIPTION
**Pull request by sakshi-arora1**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.frontend/src/utils.js [hunk 1]
```diff
@@ -94,17 +90,35 @@ export default class Utils {
...
+            }
+        }
+    }
+
+    /**
+     * Creates fields from all registered fieldCreators present inside addedElement, for given form container
+     * @param addedElement
+     * @param formContainer
+     */
+    static createFieldsForAddedElement(addedElement, formContainer) {
```
Comment (nit23uec) [inline ui.frontend/src/utils.js:117]: if its only used by instance manager, I'd suggest to create a local internal function. this has made the API public. We should only be exposing such APIs on actual use-case/demand.

File: ui.tests/test-module/specs/panelcontainer/panelcontainer.runtime.complex.spec.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
+/*******************************************************************************
+ * Copyright 2022 Adobe
```
Comment (nit23uec) [inline ui.tests/test-module/specs/panelcontainer/panelcontainer.runtime.complex.spec.js:2]: 2023

File: ui.frontend/src/utils.js [hunk 3]
```diff
@@ -94,17 +90,35 @@ export default class Utils {
...
+     * @param fieldElements
      * @param fieldCreator
-     * @param fieldElement
      * @param formContainer
      */
-    static createFormContainerField(fieldCreator, fieldElement, formContainer) {
-        let field = fieldCreator({
-            "element" : fieldElement,
-            "formContainer" : formContainer
+    static createFormContainerFields(fieldElements, fieldCreator, formContainer) {
```
Comment (nit23uec) [inline ui.frontend/src/utils.js:98]: this should be a pvt function if used only here.

File: ui.frontend/src/utils.js [hunk 4]
```diff
@@ -118,26 +132,60 @@ export default class Utils {
             console.debug("FormContainerInitialised Received", e.detail);
             let formContainer =  e.detail;
             let fieldElements = document.querySelectorAll(fieldSelector);
-            for (let i = 0; i < fieldElements.length; i++) {
-                Utils.createFormContainerField(fieldCreator, fieldElements[i], formContainer);
-            }
+            Utils.createFormContainerFields(fieldElements, fieldCreator, formContainer);
             Utils.registerMutationObserver(formContainer, fieldCreator, fieldSelector, fieldClass);
         }
+        this.#fieldCreatorSets.push({
```
Comment (nit23uec) [inline ui.frontend/src/utils.js:138]:
```
this.#fieldCreatorSets.push({
  fieldCreator,
  fieldSelector,
  fieldClass
})
```

**Review by rismehta — APPROVED** [review]

**Pull request merged**